checks-superstaq/checks_superstaq/license_header_format_.py

```diff
@@ -0,0 +1,518 @@
+    ]
+
+    return license_header_lst
+
+
+def _validate_license_header(
+    license_header_lst: list[LicenseHeader], licensee: str, license_name: str, editable: bool
+) -> bool:
+    """Returns whether there is a valid license header in a file and for each license
+    header in a file, it assigns each theiir type.
+        - VALID: if the header contains a Copyright line.
+        - OUTDATED: if the header has the correct licensee but different type.
+        - SIMILAR_LICENSE: if the header is a valid license but not from the licensee.
+        - OTHER: if the header is any other one.
+    Args:
+        license_header_lst: List of the license_header objects for each header in a file.
+        licensee: The owner of the expected license.
+        license_name: The name of the expected license.
+        editable: Whether similar licenses can be edited to include the license owner instead
+            adding the entire license to the file.
+
+    Returns: Whether there is a valid license header in a file or not.
+    """
+    valid_header_regex = re.compile(rf"(.*)Copyright(.*){licensee}")
+    outdated_header_regex = re.compile(
+        r"(.*)Copyright(.*)ColdQuanta Inc\."
+    )  # remove after first iteration
+    valid = False
+
+    for license_header in license_header_lst:
+        if (
+            re.search(valid_header_regex, license_header.license_header)
+            and license_name in license_header.license_header
+        ):
+            license_header.header_type = HeaderType.VALID
+            valid = True
+        elif re.search(outdated_header_regex, license_header.license_header):
+            # replace with re.search(valid_header_regex, license_header.license_header)
+            # and license_name not in license_header.license_header
+            license_header.header_type = HeaderType.OUTDATED
+        elif not editable or license_name not in license_header.license_header:
+            license_header.header_type = HeaderType.OTHER
+        else:
+            license_header.header_type = HeaderType.SIMILAR_LICENSE
+
+    return valid
+
+
+def _append_to_header(file: str, license_header: LicenseHeader, licensee: str) -> None:
+    """Appends licensee to existing license header.
+
+    Args:
+        file: The name/path for the file whose license header will have the licensee added to it.
+        license_header: The specific license header that the licensee is being appended to.
+        licensee: The licensee of the target license header.
+
+    Returns nothing.
+    """
+    prepend = ""
+    char_count = 0
+    with open(file, "r+") as f:
+        for line_num, line in enumerate(f):
+            char_count += len(line)
+            if (
+                "Copyright" in line
+                and license_header.start_line_num <= line_num + 1 < license_header.end_line_num
+            ):
+                if line[-2] == ",":
+                    prepend += line[:-1] + f" 2024 {licensee}.\n"
```

Comment:
silly legal q: will we need to update the year in every file on 1/1/2025? or do they stay the same until if/when we update the file?

if the former we might want to replace `2024` with e.g. `datetime.datetime.now().year`. if the latter maybe we want to make the last two digits wildcards?

Reply:
> will we need to update the year in every file on 1/1/2025?

No. Given that, we can pick whichever path is easier.